404.html

```diff
@@ -33,18 +33,13 @@
     async function loadFragment() {
             const fragmentContainer = document.getElementById('fragment-container');
             if (fragmentContainer) {
-                const path = '/fragments/404-fragment';
-                const resp = await fetch(`${path}.plain.html`);
-                if (resp.ok) {
-                    const fragmentHTML = await resp.text();
-                    fragmentContainer.innerHTML = fragmentHTML;
-                    decorateMain(fragmentContainer);
-                    await loadBlocks(fragmentContainer);
+                const fragmentHTML = await fetchFragment('/fragments/404-fragment');
```

Comment:
Cleaner approach. However, if fetchFragment returns null, you are not handling it here